### ISSUE TITLE
Create API endpoint to serve overview data 

### ISSUE THREAD

**Issue by noahmanger** [18F/openFEC#1953]
So that users can see how much money has been raised and spent by all candidates, PACs, party committees and "other" groups, aggregated by each type, provide an API endpoint that shows this data.
 
---
## Background & details
This data will be used in these charts (and potentially elsewhere):
![image](https://cloud.githubusercontent.com/assets/1696495/18855008/58804cc4-8404-11e6-8f39-9d3bb49f3996.png)

I'll leave it to others to design what the response should look like, but if it's helpful, the current charts accept an array of objects (one array for raising, one for spending) that look like this:

```
  {
    'date': '1/31/2015',
    'candidates': -12682.75,
    'pacs': 37003916.07,
    'party': 25788128.68,
    'other': 0
  }
```

@PaulClark2 has written up elsewhere the SQL that is used to generate these numbers. 

**Criteria for completion:**
- [x] The API should return a response that contains the total amount of money raised or spent by each type of group from the beginning of the cycle to the last day of each month in that cycle. 
- [x] The aggregates should be calculated on a regular basis, as new filings always come in that can change an earlier month's totals
- [x] The results should be filterable by two-year period

**Comment by PaulClark2** [comment]
Looking at Forms 3X, 3 and 3P, I think we can go back to 2007-2008. Prior to the 2007-2008 two-year period be didn't capture all of the data from the detailed summary page of reports.

I need to also look at the Form 5, 7 and 9 data.

**Comment by noahmanger** [comment]
Here's the SQL from Paul:

For candidates:
```
select
extract(month from to_date(vs.cvg_end_dt, 'YYYYMMDD')) month_order,
to_char(to_date(vs.cvg_end_dt, 'YYYYMMDD'), 'Month') as month,
extract(year from to_date(vs.cvg_end_dt, 'YYYYMMDD')) as year,
sum(nvl(vs.ttl_receipts,0) - (nvl(vs.pty_cmte_contb ,0) + nvl(vs.oth_cmte_contb ,0) + nvl(vs.offsets_to_op_exp,0) + nvl(vs.ttl_loan_repymts,0) + nvl(vs.ttl_contb_ref,0))) as adj_tr,
sum(nvl(vs.ttl_disb, 0) - (nvl(vs.tranf_to_other_auth_cmte,0) + nvl(vs.ttl_loan_repymts,0) + nvl(vs.ttl_contb_ref,0) + nvl(vs.other_disb_per,0))) as adj_td
from disclosure.v_sum_and_det_sum_report vs, disclosure.cand_cmte_linkage ccl, disclosure.candidate_summary cs
where vs.cmte_id = ccl.cmte_id
and ccl.cand_id = cs.cand_id
and ccl.cmte_tp = substr(ccl.cand_id, 1,1)
and ccl.cmte_dsgn in ('P', 'A')
and ccl.cand_election_yr in (2015, 2016)
and ccl.fec_election_yr = 2016
and cs.cand_election_yr in (2015, 2016)
and cs.fec_election_yr = 2016
and ccl.cand_id not in (select cand_id from disclosure.cand_inactive where election_yr = 2016)
and (vs.ttl_receipts > 0 or vs.ttl_disb > 0)
and vs.cvg_end_dt between 20150101 and 20160831
group by extract(month from to_date(vs.cvg_end_dt, 'YYYYMMDD')), to_char(to_date(vs.cvg_end_dt, 'YYYYMMDD'), 'Month'), extract(year from to_date(vs.cvg_end_dt, 'YYYYMMDD'))
order by extract(year from to_date(vs.cvg_end_dt, 'YYYYMMDD')), extract(month from to_date(vs.cvg_end_dt, 'YYYYMMDD'));
```

PACs:
```
select
extract(month from to_date(vs.cvg_end_dt, 'YYYYMMDD')) month_order,
to_char(to_date(vs.cvg_end_dt, 'YYYYMMDD'), 'Month') as month,
extract(year from to_date(vs.cvg_end_dt, 'YYYYMMDD')) as year,
sum(nvl(vs.ttl_receipts, 0) - (nvl(vs.pty_cmte_contb,0) + nvl(vs.oth_cmte_contb,0) + nvl(vs.offsets_to_op_exp,0) + nvl(vs.fed_cand_contb_ref_per,0) + nvl(vs.tranf_from_nonfed_acct_per,0) + nvl(vs.ttl_loan_repymts,0) + nvl(vs.ttl_contb_ref,0))) as adj_tr,
sum(nvl(vs.ttl_disb,0) - (nvl(vs.shared_nonfed_op_exp_per,0) + nvl(vs.tranf_to_other_auth_cmte,0) + nvl(vs.fed_cand_cmte_contb_per,0) + nvl(vs.ttl_loan_repymts,0) + nvl(vs.ttl_contb_ref,0) + nvl(vs.other_disb_per,0))) as adj_td
from disclosure.v_sum_and_det_sum_report vs, disclosure.cmte_valid_fec_yr cm
where vs.cmte_id = cm.cmte_id
and cm.fec_election_yr = 2016
and cm.cmte_tp in ('N', 'Q', 'O', 'V', 'W')
and cm.cmte_dsgn <> 'J'
and (vs.ttl_receipts > 0 or vs.ttl_disb > 0 or vs.debts_owed_by_cmte > 0 or vs.coh_cop >0)
and vs.cvg_end_dt between 20150101 and 20160831
group by extract(month from to_date(vs.cvg_end_dt, 'YYYYMMDD')), to_char(to_date(vs.cvg_end_dt, 'YYYYMMDD'), 'Month'), extract(year from to_date(vs.cvg_end_dt, 'YYYYMMDD'))
order by extract(year from to_date(vs.cvg_end_dt, 'YYYYMMDD')), extract(month from to_date(vs.cvg_end_dt, 'YYYYMMDD'));
```

Party committees:
```
select
extract(month from to_date(vs.cvg_end_dt, 'YYYYMMDD')) month_order,
to_char(to_date(vs.cvg_end_dt, 'YYYYMMDD'), 'Month') as month,
extract(year from to_date(vs.cvg_end_dt, 'YYYYMMDD')) as year,
sum(nvl(vs.ttl_receipts, 0) - (nvl(vs.pty_cmte_contb,0) + nvl(vs.oth_cmte_contb,0) + nvl(vs.offsets_to_op_exp,0) + nvl(vs.fed_cand_contb_ref_per,0) + nvl(vs.tranf_from_nonfed_acct_per,0) + nvl(vs.ttl_loan_repymts,0) + nvl(vs.ttl_contb_ref,0))) as adj_tr,
sum(nvl(vs.ttl_disb,0) - (nvl(vs.shared_nonfed_op_exp_per,0) + nvl(vs.tranf_to_other_auth_cmte,0) + nvl(vs.fed_cand_cmte_contb_per,0) + nvl(vs.ttl_loan_repymts,0) + nvl(vs.ttl_contb_ref,0) + nvl(vs.other_disb_per,0))) as adj_td
from disclosure.v_sum_and_det_sum_report vs, disclosure.cmte_valid_fec_yr cm
where vs.cmte_id = cm.cmte_id
and cm.fec_election_yr = 2016
and cm.cmte_tp in ('X', 'Y')
and cm.cmte_dsgn <> 'J'
and cm.cmte_id not in (select cmte_id from pclark.ref_pty_host_convention)
and vs.form_tp_cd = 'F3X'
and vs.cvg_end_dt between 20150101 and 20160831
group by extract(month from to_date(vs.cvg_end_dt, 'YYYYMMDD')), to_char(to_date(vs.cvg_end_dt, 'YYYYMMDD'), 'Month'), extract(year from to_date(vs.cvg_end_dt, 'YYYYMMDD'))
order by extract(year from to_date(vs.cvg_end_dt, 'YYYYMMDD')), extract(month from to_date(vs.cvg_end_dt, 'YYYYMMDD'));
```

Other:
```
select
extract(month from to_date(cvg_end_dt, 'YYYYMMDD')) month_order,
to_char(to_date(cvg_end_dt, 'YYYYMMDD'), 'Month') as month,
extract(year from to_date(cvg_end_dt, 'YYYYMMDD')) as year,
sum(receipts),
sum(disbursements) from
(
select
vs.cvg_end_dt,
sum(vs.TTL_CONTB) as receipts,
sum(vs.indt_exp_per) as disbursements
from DISCLOSURE.V_SUM_AND_DET_SUM_REPORT vs, DISCLOSURE.CMTE_VALID_FEC_YR cm
where vs.cmte_id = cm.cmte_id
and cm.fec_election_yr = 2016
and cm.cmte_tp = 'I'
and cm.cmte_dsgn <> 'J'
and vs.rpt_tp not in ('24', '48')
and vs.indt_exp_per > 0
and vs.cvg_end_dt between 20150101 and 20160831
group by vs.cvg_end_dt
union
select
vs.cvg_end_dt,
sum(vs.ttl_receipts) as receipts,
sum(vs.ttl_disb) as disbursements
from DISCLOSURE.V_SUM_AND_DET_SUM_REPORT vs , disclosure.cmte_valid_fec_yr cm
where vs.cmte_id = cm.cmte_id
and cm.fec_election_yr = 2016
and vs.cvg_end_dt between 20150101 and 20160831
and vs.form_tp_cd = 'F9'
group by vs.cvg_end_dt
union
select
cvg_end_dt,
0 as receipts,
SUM(TTL_COMMUNICATION_COST) as disbursements
from DISCLOSURE.V_SUM_AND_DET_SUM_REPORT vs, DISCLOSURE.CMTE_VALID_FEC_YR cm
where vs.cmte_id = cm.cmte_id
and cm.fec_election_yr = 2016
and cm.cmte_tp = 'C'
and cm.cmte_dsgn <> 'J'
and vs.TTL_COMMUNICATION_COST > 0
and vs.cvg_end_dt between 20150101 and 20160831
group by vs.cvg_end_dt
)
group by extract(month from to_date(cvg_end_dt, 'YYYYMMDD')), to_char(to_date(cvg_end_dt, 'YYYYMMDD'), 'Month'), extract(year from to_date(cvg_end_dt, 'YYYYMMDD'))
order by extract(year from to_date(cvg_end_dt, 'YYYYMMDD')), extract(month from to_date(cvg_end_dt, 'YYYYMMDD'));
```

**Comment by PaulClark2** [comment]
Does it make sense to rewrite this SQL to use candidate summary and committee summary, where possible?

**Comment by LindsayYoung** [comment]
@PaulClark2 Yes, I am working on that and will get back to you so you can check it

**Comment by LindsayYoung** [comment]
Hey @PaulClark2 wanted to confirm a few things and have some questions for you

Confirming:
- Joint pacs aren't included anywhere to prevent double-counting
- Committee code Z isn't used
- tranf_to_other_auth_cmte_per in fec_vsum_f3pis the same as tranf_to_other_auth_cmte in your script

Questions:
- We don't seem to be counting delegate committees, don't think they will make a big difference, but we can add them to other, to be more accurate
- There is a reference to pclark.ref_pty_host_convention in the sql you gave me, is there a way for us to replicate that? Should we worry about it?


I did my first pass of the sql with postgres using existing tables and views. I need to check the data still but it looks functional. Here is the file if you want to dig in:
https://github.com/18F/openFEC/blob/feature/chart-endpoints/data/sql_updates/large_aggregates.sql

**Comment by noahmanger** [comment]
Resolved by #1964 . We're continuing to QA the data in another issue.
